EIPS/eip-7934.md

```diff
@@ -0,0 +1,77 @@
+---
+eip: 7934
+title: RLP Execution Block Size Limit
+description: Introduce a protocol-level cap on the maximum RLP-encoded block size to 10 MB, including a 512 KB margin for beacon block size.
+author: Giulio Rebuffo (@Giulio2002), Ben Adams (@benaadams), Storm Slivkoff (@sslivkoff)
+discussions-to: https://ethereum-magicians.org/t/eip-7934-add-bytesize-limit-to-blocks/23589
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-04-16
+---
+
+## Abstract
+
+This proposal introduces a protocol-level cap on the maximum RLP-encoded execution block size to 10 megabytes (MB), which includes a margin of 512 KB to account for beacon block sizes.
+
+## Motivation
+
+Currently, Ethereum does not enforce a strict upper limit on the encoded size of blocks. This lack of constraint can result in:
+
+1. **Network Instability**: Extremely large blocks slow down propagation and increase the risk of temporary forks and reorgs.
+2. **DoS Risks**: Malicious actors could generate exceptionally large blocks to disrupt network performance.
+
+Additionally, blocks exceeding 10 MB are not propagated by the consensus layer's (CL) gossip protocol, potentially causing network fragmentation or denial-of-service (DoS) conditions.
```

Comment:
Looks like EL devp2p will disconnect peers that gossip messages over 10MiB also

> The practical limits for the eth protocol are lower, typically 10 MiB. If a received message is larger than the limit, the peer should be disconnected.

https://github.com/ethereum/devp2p/blob/master/caps/eth.md#ethereum-wire-protocol-eth